python/packages/ai/teams/ai/prompts/prompt_manager.py

```diff
@@ -20,18 +20,19 @@
 
 SK_CONFIG_FILE_NAME = "config.json"
 SK_PROMPT_FILE_NAME = "skprompt.txt"
+StateT = TypeVar("StateT", bound=TurnState)
 
 
-class PromptManager:
+class PromptManager(Generic[StateT]):
     """
     Prompt manager used by action planner internally
     """
 
     _prompts_folder: str
     _templates: Dict[str, PromptTemplate] = {}
-    _functions: Dict[str, Callable[[TurnContext, TurnState], str]] = {}
+    _functions: Dict[str, Callable[[TurnContext, StateT], Awaitable[str]]] = {}
 
-    def __init__(self, prompts_folder: str) -> None:
+    def __init__(self, prompts_folder: str = "prompts") -> None:
```

Comment:
Can you share why we need to have default value for prompts_folder?

Reply:
see comment above about defaults